Update release notes (#352)
vmware-tanzu/sources-for-knative: RELEASE.md

directly push to the upstream repo).

Create a temporary clone of the repo, e.g. to `/tmp`. Then create the `tag` and

> push to origin. Since knative 1.0 the releases in this repo still are versioned in the `0.x.y` format, and the tags corresponds to the knative tags `knative-v1.x`. This starts from `v0.27.0` == `knative-v1.0` and so on. For example, to create a release called `v0.27.0`, you would run the following commands:

Reviewer: the knative prefixed tags are still `knative-vx.x.x`, not `knative-v1.0` like you have here

> push to origin. Since knative 1.0 the releases in this repo still are versioned in the `v0.x.y` format, and the tags corresponds to the knative tags `knative-v1.x.y`. This starts from `v0.27.0` == `knative-v1.0.0` and so on. For example, to create a release called `v0.27.0`, you would run the following commands:

Reviewer 2: Just to verify: if I create tag `v0.28.0` that would correspond to `knative-v1.1.0` then?

Reviewer 2: Perhaps another example (briefly) to describe the mapping for when knative-1.1.0 is to be created would be neat.

```shell
cd /tmp
git clone git@github.com:vmware-tanzu/sources-for-knative.git
cd sources-for-knative

# checkout the branch created above
git checkout release-1.0

# associate the tag with the branch
git tag -a v0.27.0 -m "Release v0.27.0"
# since knative version 1.0 we need to create an aditional tag that matches the
# knative-v1.x.y tag format
git tag -a knative-v1.0.0 -m "Release knative-v1.0.0"

git push origin v0.27.0

> git push origin knative-v1.0.0

Reviewer 2: PR #353 adds `knative-*` as push target for creating the release. Are we now triggering two release runs for the same commit by pushing two tags?

Author: Was a mistake from my part on that #353 PR, now it's good

directly push to the upstream repo).

Create a temporary clone of the repo, e.g. to `/tmp`. Then create the `tag` and

> push to origin. Since knative 1.0 the releases in this repo still are versioned in the `v0.x.y` format, and the tags corresponds to the knative tags `knative-v1.x.y`.

Reviewer 2: ```suggestion
push to origin. Despite Knative 1.0, the release tags in this repo are still versioned in the `v0.x.y` format and correspond to the Knative tags `knative-v1.x.y`.
```

> This starts from `v0.27.0` == `knative-v1.0.0` and each MINOR version addition in the knative-v1.x tag corresponds to the same addition on the 0.x tag.

Reviewer 2: ```suggestion
This starts from `v0.27.0` == `knative-v1.0.0` and each MINOR version addition in the `knative-v1.x` tag corresponds to the same addition on the `0.x` tag.
```

> PATCH versions are ignored in the branch name, in case a release version MUST be patched, just redo the release in the same release branch pointing to the correct commit and with the correct tag, i.e:

Reviewer 2: ```suggestion
PATCH versions are ignored in the branch name. In case a release version MUST be patched, just redo the release in the same release branch pointing to the correct commit and with the correct tag, i.e:
```

| Branch       | Knative Tag  | Normal Tag       |
|--------------|--------------|------------------|
| release-v1.1 | `v0.27.0`    | `knative-v1.0.x` |
| release-v1.1 | `v0.28.0`    | `knative-v1.1.x` |
| release-v1.1 | `v0.28.1`    | `knative-v1.1.x` |
| release-v1.2 | `v0.29.0`    | `knative-v1.2.x` |
| release-v1.3 | `v0.30.0`    | `knative-v1.3.x` |
and so on...

> A practical example to create a release called `v0.27.0`, you would run the following commands:

Reviewer 2: ```suggestion
As a practical example, to create a release called `v0.27.0` you would run the following commands:
```

```shell
cd /tmp
git clone git@github.com:vmware-tanzu/sources-for-knative.git
cd sources-for-knative

# checkout the branch created above
git checkout release-1.0

# associate the tag with the branch
git tag -a v0.27.0 -m "Release v0.27.0"

> # since knative version 1.0 we need to create an aditional tag that matches the

Reviewer 2: nit: add empty line before

```suggestion

# since knative version 1.0 we need to create an aditional tag that matches the
```

# knative-v1.x.y tag format
git tag -a knative-v1.0.0 -m "Release knative-v1.0.0"

git push origin v0.27.0
git push origin knative-v1.0.0
```

**Notes:**
- To trigger the release workflow you cannot push more than 2 tags at a time.
- After this process finish, don't forget to go to the [Release Section](https://github.com/vmware-tanzu/sources-for-knative/releases), edit the latest release and deselect the `this is a pre-release` option.

## Release Notes

Release notes can be generated via the corresponding Github Actions

> [Workflow](.github/workflows/knative-release-notes.yaml). The workflow can be

Reviewer 2: ```suggestion
[workflow](.github/workflows/knative-release-notes.yaml). The workflow can be
```

directly push to the upstream repo).

Create a temporary clone of the repo, e.g. to `/tmp`. Then create the `tag` and
push to origin. Despite Knative 1.0, the release tags in this repo are still versioned in the `v0.x.y` format and correspond to the Knative tags `knative-v1.x.y`.
This starts from `v0.27.0` == `knative-v1.0.0` and each MINOR version addition in the `knative-v1.x.y` tag corresponds to the same addition on the `0.x.z` tag.

PATCH versions are ignored in the branch name. In case a release version MUST be patched, just redo the release in the same release branch pointing to the correct commit and with the correct tag, i.e:

| Branch      | Normal Tag  | Knative Tag      |
|-------------|-------------|------------------|

> | release-1.1 | `v0.27.0`   | `knative-v1.0.x` |

Reviewer 2: Sorry for bugging you (seems you missed that comment earlier), but according to your description above, this should be:

```suggestion
| release-1.0 | `v0.27.0`   | `knative-v1.0.x` |
```

Author: You are absolutely right 👍🏼 my b

> | release-v1.1 | `v0.27.0`    | `knative-v1.0.x` |

Reviewer 2: This does not match the branch description created above (`release-1.0`)

| release-1.1 | `v0.28.0`   | `knative-v1.1.x` |
| release-1.1 | `v0.28.1`   | `knative-v1.1.x` |
| release-1.2 | `v0.29.0`   | `knative-v1.2.x` |
| release-1.3 | `v0.30.0`   | `knative-v1.3.x` |
and so on...

As a practical example, to create a release called `v0.27.0` you would run the following commands:

```shell
cd /tmp
git clone git@github.com:vmware-tanzu/sources-for-knative.git
cd sources-for-knative

# checkout the branch created above
git checkout release-1.0

# associate the tag with the branch
git tag -a v0.27.0 -m "Release v0.27.0"

# since knative version 1.0 we need to create an aditional tag that matches the
# knative-v1.x.y tag format
git tag -a knative-v1.0.0 -m "Release knative-v1.0.0"

> git push origin v0.27.0

Reviewer 2: nit: I would make this release and the knative tag env variables since it's easy to mess this up when copying/pasting doing manual.

We usually follow this pattern (in other non-kn repos) to ease releasing for the operator, e.g. in [`govmomi`](https://github.com/vmware/govmomi/blob/master/RELEASE.md): 

```console
RELEASE=v0.27.0
KN_RELEASE=knative-v1.0.0
```

and then reuse these across the `tag` and `push` commands, e.g.

```console
git tag -a ${RELEASE} -m "Release ${RELEASE}"
git tag -a ${KN_RELEASE} -m "Release ${KN_RELEASE}"

git push origin ${RELEASE}
git push origin ${KN_RELEASE}
```

```

**Notes:**
- To trigger the release workflow you cannot push more than 2 tags at a time.

> - After this process finish, don't forget to go to the [Release Section](https://github.com/vmware-tanzu/sources-for-knative/releases), edit the latest release and deselect the `this is a pre-release` option.

Reviewer 2: IMHO we should fix .goreleaser.yaml and drop this line

## Release Notes

Release notes can be generated via the corresponding Github Actions
[Workflow](.github/workflows/knative-release-notes.yaml). The workflow can be
triggered via manual invocation (`workflow_dispatch`)

> [Here](https://github.com/vmware-tanzu/sources-for-knative/actions/workflows/knative-release-notes.yaml).

Reviewer 2: nit: really uppercase?

PATCH versions are ignored in the branch name. In case a release version MUST be patched, just redo the release in the same release branch pointing to the correct commit and with the correct tag, i.e:

| Branch      | Knative Tag  | Normal Tag       |
|-------------|--------------|------------------|

> | release-1.1 | `v0.27.0`    | `knative-v1.0.x` |

Reviewer 2: Should be `release-1.0` ?

cd sources-for-knative

# checkout the branch created above

> git checkout release-1.1

Reviewer 2: this does not match the above
